pandas/core/arrays/datetimelike.py

```diff
@@ -1263,7 +1593,8 @@ def _ensure_datetimelike_to_i8(other, to_utc=False):
 
     if lib.is_scalar(other) and isna(other):
         return iNaT
-    elif isinstance(other, (PeriodArray, ABCIndexClass)):
+    elif isinstance(other, (PeriodArray, ABCIndexClass,
+                            DatetimeLikeArrayMixin)):
```

Comment:
It looks like this is needed in order for `round` to work corretly on tz-aware DatetimeArray.  I'll be making a PR with this edit and a test before long.